src/audio/codec_adapter/codec_adapter.c

```diff
@@ -526,26 +466,16 @@ static int codec_adapter_set_params(struct comp_dev *dev, struct sof_ipc_ctrl_da
 		return 0;
 
 	/* config fully copied, now load it */
-	switch (type) {
-	case MODULE_CFG_SETUP:
-		ret = load_setup_config(dev, md->runtime_params, size);
-		if (ret)
-			comp_err(dev, "codec_adapter_set_params(): error %d: load of setup config failed.",
-				 ret);
-		else
-			comp_dbg(dev, "codec_adapter_set_params() load of setup config done.");
-		break;
-	case MODULE_CFG_RUNTIME:
-		ret = module_load_config(dev, md->runtime_params, size, MODULE_CFG_RUNTIME);
-		if (ret) {
-			comp_err(dev, "codec_adapter_set_params() error %d: load of runtime config failed.",
-				 ret);
-			break;
-		}
+	ret = module_load_config(dev, md->runtime_params, size, type);
+	if (ret) {
+		comp_err(dev, "codec_adapter_set_params(): error %d: config failed", ret);
```

Comment:
actually `module_load_config()` already prints an error in all error cases. And btw you can replace `return ret;` with an explicit `return 0;` in the success case there.

Reply:
I've removed the error but I left the return ret because it makes it easier to read

Reply:
"easier to read" is soooo subjective ;-) But sure, it must not be easier to read to me :-) Although it kind of seems easier to me: `return 0;` you know the value immediately, and you know it's success. With `return ret;` you have to check what `ret` is, then you see - aha, I have `if (ret) return ret;` above, so, here it can only be 0, so, we're returning 0 here. That seems like a longer logical chain to me :-)

Reply:
I'm still not getting it: we have 
```
if (md->runtime_params)
rfree(md->runtime_params);
md->runtime_params = NULL;
return ret;
```
that is common for both error and success. Where do you want the return 0 here?